File: Android/app/src/main/java/com/kite/zmusic/config/NcmApiConfig.kt
package com.kite.zmusic.config

import com.kite.zmusic.BuildConfig

/**
 * 网易云兼容 API（NeteaseCloudMusicApi）基地址。
 *
 * 编译期默认：`BuildConfig.NCM_API_BASE_URL`（可由 `Android/local.properties` 的
 * `ncm.api.base.url` 覆盖）。运行期可由用户配置覆盖并持久化。
 */
object NcmApiConfig {
    val defaultBaseUrl: String = BuildConfig.NCM_API_BASE_URL.trimEnd('/')

    @Volatile
    private var runtimeBaseUrl: String? = null

    val baseUrl: String
        get() = runtimeBaseUrl?.takeIf { it.isNotBlank() } ?: defaultBaseUrl

    fun setRuntimeBaseUrl(url: String) {
        runtimeBaseUrl = url.trim().trimEnd('/').takeIf { it.isNotEmpty() }
    }
}

File: Windows/Config/NcmApiConfig.cs
namespace ZMusic.Config;

/// <summary>
/// NeteaseCloudMusicApi base URL. Default matches Android; override via env ZMUSIC_NCM_API_BASE_URL.
/// </summary>
public static class NcmApiConfig
{
    public const string DefaultBaseUrl = "http://120.27.244.170:3000";

    private static string? _runtimeBaseUrl;

    public static string BaseUrl
    {
        get
        {
            if (!string.IsNullOrWhiteSpace(_runtimeBaseUrl))
            {
                return _runtimeBaseUrl.TrimEnd('/');
            }

            var fromEnv = Environment.GetEnvironmentVariable("ZMUSIC_NCM_API_BASE_URL");
            if (!string.IsNullOrWhiteSpace(fromEnv))
            {
                return fromEnv.Trim().TrimEnd('/');
            }

            return DefaultBaseUrl;
        }
    }

    public static void SetRuntimeBaseUrl(string? url)
    {
        _runtimeBaseUrl = string.IsNullOrWhiteSpace(url)
            ? null
            : url.Trim().TrimEnd('/');
    }
}
